wsmatrix@martin.zurowietz.de/Settings.js

```diff
@@ -1,9 +1,10 @@
 const WsMatrix = imports.misc.extensionUtils.getCurrentExtension();
+const GObject = imports.gi.GObject;
 const Gio = imports.gi.Gio;
 const GioSSS = Gio.SettingsSchemaSource;
 
-var Settings = class Settings extends Gio.Settings {
-  constructor(schema) {
+var Settings = GObject.registerClass(class Settings extends Gio.Settings {
```

Comment:
please fix formatting here
```Javascript
var Settings = GObject.registerClass(
class Settings extends Gio.Settings {
```

Reply:
unfortunately, it's not documented, but it's mainly similar to [gjs style](https://wiki.gnome.org/Projects/GnomeShell/Gjs_StyleGuide), with these differences 
1- indentation is 3 spaces instead of 4 (and should be spaces, not tabs)
2- if/for statements must always have curly braces (gjs recommends not having curly braces when possible)

I am not sure if there are other guidelines, maybe @mzur should document this when he has time.

Reply:
Well, it is two spaces in Settings and four - in WsmatrixThumbnail. I've just indented classes inside registerClass().